modules/cli/src/main/scala/scala/cli/commands/Update.scala

```diff
@@ -50,48 +57,97 @@ object Update extends ScalaCommand[UpdateOptions] {
     }
   }
 
-  def update(options: UpdateOptions, scalaCliBinPath: os.Path) = {
-    val currentVersion = {
-      val res = os.proc(scalaCliBinPath, "version").call(cwd = os.pwd, check = false)
-      if (res.exitCode == 0)
-        res.out.text().trim
-      else
-        "0.0.0"
+  /** Either an error or a boolean indicating whether there are duplicates for scala-cli on PATH
+    */
+  lazy val hasPathDuplicates: Either[String, Boolean] = either {
+    val instances = value(pathInstances)
+    instances.length > 1
+  }
+
+  lazy val updateInstructions: String = {
+    val base = s"""Your Scala CLI version is outdated. The newest version is $newestScalaCliVersion
+                  |It is recommended that you update Scala CLI through the same tool or method you used for its initial installation for avoiding the creation of outdated duplicates.""".stripMargin
+    if (pathInstances.isRight)
+      pathDuplicatesWarning.right.get
+    else base
+  }
+
+  /** Either an error or the warning about more than one instance of scala-cli on PATH
+    */
+  lazy val pathDuplicatesWarning: Either[String, String] = either {
```

Comment:
Also there is a hey problem with this feature namely on Ubunbu I've got scala-cli installed in 2 places:

```
⋊> ~ type -ap scala-cli                                               
/usr/bin/scala-cli
/bin/scala-cli
```

I was using provided installer and both are a scripts that calls the same scala-cli instance on `/usr/share/scala/scala-cli` - so I would get the warning everytime and it would be really annoying. I think we need to discuss how this should work on our standup.

Reply:
So if these two results are both fed to `ls -l` it would show that they both refer to the same location. So this requires one extra step of feeding to `ls -l` and then parsing the returned result to make sure these are real duplicates.
The warning would only be shown on update, about and version commands.

Reply:
regarding the `Either`, for example, if the operating system is windows, it cannot calculate whether there is any duplicates on path or not. So, in those cases, it should not show a warning to the user.

Reply:
I removed the duplicate finding code!